# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/internal_types.rs
# Commit: 15da94c6a7b1
# Full Hash: 15da94c6a7b19f971d3dffaa4cab8edb8e41cb62
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2020-12-15 09:29:54
# Regressor Bug: 1682365
# File Overlap Count: 1
# Description:
#   Bug 1682365 - Pt 2 - Remove texture array usage from render task graph. r=nical
#   
#   With this change, all color/alpha intermediate surfaces are individual
#   textures, rather than texture arrays.
#   
# ==============================================================================

diff -r 72ddac486037 -r 15da94c6a7b1 gfx/wr/webrender/src/internal_types.rs
--- a/gfx/wr/webrender/src/internal_types.rs	Tue Dec 15 01:46:51 2020 +0200
+++ b/gfx/wr/webrender/src/internal_types.rs	Mon Dec 14 21:48:41 2020 +0000
@@ -265,7 +265,7 @@
     /// Equivalent to `None`, allowing us to avoid using `Option`s everywhere.
     Invalid,
     /// An entry in the texture cache.
-    TextureCache(CacheTextureId, ImageBufferKind, Swizzle),
+    TextureCache(CacheTextureId, Swizzle),
     /// An external image texture, mananged by the embedding.
     External(DeferredResolveIndex, ImageBufferKind),
     /// Select a dummy 1x1 white texture. This can be used by image
@@ -276,12 +276,12 @@
 impl TextureSource {
     pub fn image_buffer_kind(&self) -> ImageBufferKind {
         match *self {
-            TextureSource::TextureCache(_, image_buffer_kind, _) => image_buffer_kind,
+            TextureSource::TextureCache(..) => ImageBufferKind::Texture2D,
 
             TextureSource::External(_, image_buffer_kind) => image_buffer_kind,
 
             // Render tasks use texture arrays for now.
-            TextureSource::Dummy => ImageBufferKind::Texture2DArray,
+            TextureSource::Dummy => ImageBufferKind::Texture2D,
 
             TextureSource::Invalid => ImageBufferKind::Texture2D,
         }